packages/react-components/react-dialog/src/components/Dialog/useDialog.ts

```diff
@@ -64,7 +64,7 @@ export const useDialog_unstable = (props: DialogProps): DialogState => {
     dialogTitleId: useId('dialog-title-'),
     isNestedDialog: useHasParentContext(DialogContext),
     dialogRef: focusRef,
-    modalAttributes: modalType !== 'non-modal' ? modalAttributes : undefined,
+    modalAttributes,
```

Comment:
So, now we will pass `modalAttributes` always which are:

```ts
  const { modalAttributes, triggerAttributes } = useModalAttributes({
    trapFocus: modalType !== 'non-modal',
    legacyTrapFocus: !inertTrapFocus,
  });
```

Should it be then following?

```diff
  const { modalAttributes, triggerAttributes } = useModalAttributes({
    trapFocus: modalType !== 'non-modal',
-   legacyTrapFocus: !inertTrapFocus,
+   legacyTrapFocus: modalType !== 'non-modal' && !inertTrapFocus,
  });
```

Reply:
> Should it be then following?

No need actually, `legacyTrapFocus` is just a flag that is used in `useModalAttributes` together with `trapFocus` in a logic and condition, so `!inertTrapFocus` should work just fine